File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.java
import java.util.*;
class Solution {
    public static long solution(int cap, int n, int[] deliveries, int[] pickups) {
    List<int[]> idps = new ArrayList<>();
    for (int i = 0; i < n; i++) {
        int d = deliveries[i];
        int p = pickups[i];
        if (d > 0 || p > 0) {
            int[] pair = new int[]{i+1, d, p};
            idps.add(pair);
        }
    }
    int delivery = 0;
    int pickup = 0;
    long answer = 0;
    while (!idps.isEmpty()) {
        int[] idp = idps.remove(idps.size()-1);
        int i = idp[0];
        int d = idp[1];
        int p = idp[2];
        delivery += d;
        pickup += p;
        while (delivery > 0 || pickup > 0) {
            delivery -= cap;
            pickup -= cap;
            answer += i*2;
        }
    }
    return answer;

    }
}

File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    idps = [(i, d, p) for i, (d, p) in enumerate(zip(deliveries, pickups), 1) if d or p]
    delivery = 0
    pickup = 0
    answer = 0
    while idps:
        i, d, p = idps.pop()
        # -2 -4
        # -1 0
        # 2, 0
        # -2, -4
        # -2, -1
        # 1-, -1

        delivery += d 
        pickup += p
        while delivery > 0 or pickup > 0:
            delivery -= cap
            pickup -= cap
            answer += i*2
    return answer